compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1548,11 +1548,24 @@ private void validateXMLConstraintType(BType type, Location pos) {
             return;
         }
 
+        if (constrainedTag == TypeTags.INTERSECTION) {
```

Comment:
Still won't work for something like

```ballerina
xml<(xml:Element|xml:Comment) & readonly> _ = xml ``;
```

Reply:
IMO, this isn't where we should check for intersection type. After L1543, you can check if the constraint is an intersection type and set the effective type to `constraintType`. That way the union logic will also work as is and you wouldn't need to special case like this.

Reply:
Yes. I will change it.

Reply:
Updated -  92b1e95bcd91c8dc89775d0f23f3a91ef0236f75